embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/api/ApiClientImpl.kt

```diff
@@ -27,13 +27,14 @@
     override fun executeGet(request: ApiRequest): ApiResponse<String> {
         var connection: EmbraceConnection? = null
 
-        try {
+        return try {
             connection = request.toConnection()
             setTimeouts(connection)
             connection.connect()
-            return executeHttpRequest(connection)
+            val response = executeHttpRequest(connection)
+            response
         } catch (ex: Throwable) {
-            throw IllegalStateException(ex.localizedMessage ?: "", ex)
+            ApiResponse.Error(IllegalStateException(ex.localizedMessage ?: "", ex))
```

Comment:
Just add the @Suppress to this method, as this is something we'll want to change. I don't think it's an IllegalStateException at all, and I'm not sure why we're even wrapping it. So we can count them together? I think we lose some data fidelity by munging them all together.

Reply:
We don't need a Suppress here, as the warning to not throw an IllegalStateException is in EmbraceApiService because it was not in the context of a try catch. 
I agree with you in that we are unnecessary wrapping exceptions into a IllegalStateException, not sure why we were doing this, that's why we better investigate if there is a reason and address it in another PR.